### ISSUE TITLE
Fix intermittent custom_notification test failure

### ISSUE THREAD

**Issue by redshiftzero** [freedomofpress/securedrop#1736]
The `custom_notification` test is occasionally failing in the staging environment. When I tested #1731 it did not fail (using docker and AWS) but I have seen it fail with staging provisioned locally using vagrant. @conorsch mentioned that this is a test failure only occurring in the vagrant environment. We should resolve this for 0.4.

**Comment by conorsch** [comment]
Working on this now. Looks like the problem is the nested group memberships as created by Vagrant for the INI-style inventory file. We should be using host_vars to override group_vars, e.g. for app-staging. Will reorg some of the vars and confirm passing tests locally.

**Comment by conorsch** [comment]
The vars story is kind of a mess right now. Thanks to the recent Admin-related tooling, we have a much more stable configuration flow, so I'm going to shove most of the top-level vars into role defaults. They'll still be configurable, but the majority aren't being overridden, so let's get them out of sight. Setting host-specific overrides, e.g. for `custom_notification_text`, will then be a much more straightforward process.

**Comment by redshiftzero** [comment]
I'm seeing this failure (locally - vagrant + VirtualBox on Mac OS X) on current develop, so reopening for now.

**Comment by conorsch** [comment]
@redshiftzero Cannot reproduce. Here's what I ran:

```
#!/bin/bash
# Steps to reproduce #1736:
# https://github.com/freedomofpress/securedrop/issues/1736
set -x

# Provision fresh VMs for testing
vagrant up --no-provision build
vagrant destroy -f /staging/
vagrant up /staging/

# Reload staging machines so config tests pass.
vagrant reload /staging/

# Wait for VMs to come back up...
sleep 30

# Run config tests (never hurts to be sure)
for vm in build app-staging mon-staging ; do
    ./testinfra/test.py "${vm}"
done

# Run app tests (this is where failure will happen)
vagrant ssh app-staging -c "sudo su -c 'pytest -v /var/www/securedrop/tests'"

# Print out version info
vagrant --version
vagrant plugin list
ansible --version
git rev-parse HEAD
```

Exits non-zero for me. The version info reported by the final block displays:

```
+ vagrant --version
Vagrant 1.9.5
+ vagrant plugin list
vagrant-hostmanager (1.8.6)
vagrant-libvirt (0.0.40)
  - Version Constraint: > 0
vagrant-mutate (1.2.0)
vagrant-share (1.1.8, system)
+ ansible --version
ansible 2.2.1.0
  config file = /home/conor/freedomofpress/securedrop/ansible.cfg
  configured module search path = Default w/o overrides
+ git rev-parse HEAD
4ab6269f50f59655bb4522a25ba1b6d7014cf1c9
```

It'd be super helpful if you executed that script as-is and posted a gist of the output, if anything fails.

**Comment by redshiftzero** [comment]
Thanks @conorsch :gem:. I ran the script you provided and everything passed. So since the test passed in staging VMs and not the development VM, I investigated a bit and it looks like in some of the recent Ansible shakeup (converting the group vars to host vars for the development VM in #1838) the `custom_notification` "This is an insecure SecureDrop Development server for testing ONLY. Do NOT submit documents here." no longer appeared in the development VM. This caused the test failure, but only locally and only in the development VM. Anyway, it's back in the host vars in #1921, and all tests are passing for me locally in the development VM with that addition so this is good to close if that looks OK

**Comment by conorsch** [comment]
@redshiftzero Ahh thanks, I was only targeting app-test. Will reproduce the failure and confirm fix in #1921 for review!

**Comment by conorsch** [comment]
Troublingly I wasn't able to reproduce the failure on the development VM, either. Here's the script I ran:

```
#!/bin/bash
# Steps to reproduce #1736:
# https://github.com/freedomofpress/securedrop/issues/1736
set -x

# Provision fresh VMs for testing
vagrant destroy -f /dev/
vagrant up /dev/

# Wait for VMs to come back up...
sleep 30

# Run config tests (never hurts to be sure)
./testinfra/test.py development

# Run app tests (this is where failure will happen)
vagrant ssh development -c "pytest -v /vagrant/securedrop/tests"

# Print out version info
vagrant --version
vagrant plugin list
ansible --version
git rev-parse HEAD
```

And version output:

```
+ vagrant --version
Vagrant 1.9.5
+ vagrant plugin list
vagrant-hostmanager (1.8.6)
vagrant-libvirt (0.0.40)
  - Version Constraint: > 0
vagrant-mutate (1.2.0)
vagrant-share (1.1.8, system)
+ ansible --version
ansible 2.2.1.0
  config file = /home/conor/freedomofpress/securedrop2/ansible.cfg
  configured module search path = Default w/o overrides
+ git rev-parse HEAD
d8283cd972908059960bb4fc9d91e3d5acfd2b37
```

Everything passes green across the board. Mostly I'm disturbed that you and I are seeing differently results, but will proceed with review of #1921 anyway.
